atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/ClientLockWatchEventLogImpl.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api.watch;
+
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Streams;
+import com.palantir.lock.watch.ClientLockWatchEventLog;
+import com.palantir.lock.watch.ClientLockWatchSnapshotUpdater;
+import com.palantir.lock.watch.ClientLogEvents;
+import com.palantir.lock.watch.IdentifiedVersion;
+import com.palantir.lock.watch.LockWatchCreatedEvent;
+import com.palantir.lock.watch.LockWatchEvent;
+import com.palantir.lock.watch.LockWatchStateUpdate;
+import com.palantir.logsafe.Preconditions;
+
+final class ClientLockWatchEventLogImpl implements ClientLockWatchEventLog {
+    private static final boolean INCLUSIVE = true;
+
+    private final ClientLockWatchSnapshotUpdater snapshotUpdater;
+    private final NavigableMap<Long, LockWatchEvent> eventMap = new TreeMap<>();
```

Comment:
I would maybe again suggest splitting a small datastructure class that has the API you need here.

Reply:
I personally disagree with this one. The other split made sense to me as it was strongly coupled (i.e. two maps), but I don't think we get much from wrapping this map in another class (other than a longer diff!). Unless you feel strongly about this I would rather keep as-is.